ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStoreFactory.java

```diff
@@ -67,44 +68,32 @@ public AccountStatsMySqlStoreFactory(VerifiableProperties verifiableProperties,
    * @throws SQLException
    */
   @Override
-  public AccountStatsStore getAccountStatsStore() throws SQLException {
+  public AccountStatsStore getAccountStatsStore() throws Exception {
     Map<String, List<DbEndpoint>> dcToMySqlDBEndpoints = getDbEndpointsPerDC(accountStatsMySqlConfig.dbInfo);
     List<DbEndpoint> dbEndpoints = dcToMySqlDBEndpoints.get(localDC);
     if (dbEndpoints == null || dbEndpoints.size() == 0) {
       throw new IllegalArgumentException("Empty db endpoints for datacenter: " + localDC);
     }
-    if (accountStatsMySqlConfig.enableRewriteBatchedStatement) {
-      dbEndpoints = setRewriteBatchedStatements(dbEndpoints);
-    }
-    try {
-      dbEndpoints.forEach(ep -> logger.info("DBUrl: {}", ep.getUrl()));
-      return new AccountStatsMySqlStore(accountStatsMySqlConfig, dbEndpoints, localDC, clusterName, hostname,
-          localBackupFilePath, hostnameHelper, registry);
-    } catch (SQLException e) {
-      logger.error("Account Stats MySQL store creation failed", e);
-      throw e;
-    }
+    dbEndpoints.forEach(ep -> logger.info("DBUrl: {}", ep.getUrl()));
+    return new AccountStatsMySqlStore(accountStatsMySqlConfig, buildDataSource(dbEndpoints.get(0)), clusterName,
+        hostname, localBackupFilePath, hostnameHelper, registry);
   }
 
-  /**
-   * Set rewriteBatchedStatements for the db url if not set yet.
-   * @param endpoints The list of {@link DbEndpoint}s.
-   * @return The modified lit of {@link DbEndpoint}s.
-   */
-  static List<DbEndpoint> setRewriteBatchedStatements(List<DbEndpoint> endpoints) {
-    List<DbEndpoint> result = new ArrayList<>();
-    for (DbEndpoint ep : endpoints) {
-      if (ep.getUrl().contains(REWRITE_BATCHED_STATEMENTS_LITERAL)) {
-        result.add(ep);
-      } else {
-        String sep = ep.getUrl().contains("?") ? "&" : "?";
-        String newUrl = new StringBuilder(ep.getUrl()).append(sep)
-            .append(REWRITE_BATCHED_STATEMENTS_LITERAL)
-            .append("=true")
-            .toString();
-        result.add(new DbEndpoint(newUrl, ep.getDatacenter(), ep.isWriteable(), ep.getUsername(), ep.getPassword()));
-      }
-    }
-    return result;
+  private HikariDataSource buildDataSource(DbEndpoint dbEndpoint) {
```

Comment:
not for this PR, but in the future, we can probably put this in a util class since it is pretty similar to the setup code used in named blobs.